src/Controller/UserController.php

```diff
@@ -59,63 +62,82 @@ public function viewUser(User $user = null, UserRepository $userRepository, Seri
-     * @return User
+     * @return View
      */
-    public function newUser(Request $request, EntityManagerInterface $manager, ClientRepository $repository)
+    public function newUser(User $user, EntityManagerInterface $manager, ValidatorInterface $validator, $clientId,
+                            ClientRepository $repository)
     {
-        $data = json_decode($request->getContent(), true);
-
-        $clientId = $data['client']['id'];
         $client = $repository->findClient($clientId);
+        $user->setClient($client[0]);
 
-        if ($client) {
-            $user = new User();
-            $user->createUserObject($data);
-            $user->setClient($client[0]);
+        $validatorResults = $validator->validate($user, null, null);
+        if(count($validatorResults) > 0){
+            return $this->view($validatorResults, Response::HTTP_BAD_REQUEST);
+        }
 
-            $manager->persist($user);
-            $manager->flush();
+        $manager->persist($user);
+        $manager->flush();
 
-            return $user;
-        }
+        return $this->view($user, Response::HTTP_CREATED);
     }
 
     /**
      * @Rest\Put(
      *     path = "/modifyuser/{userId}",
      *     name = "modify_user")
-     * @param Request $request
-     * @param EntityManagerInterface $manager
-     * @param ClientRepository $repository
-     * @param UserRepository $userRepository
-     * @return User
+     * @ParamConverter("user", converter="fos_rest.request_body")
+     * @param $userId
+     * @param UserRepository $repository
+     * @return View
      */
-    public function modifyUser(Request $request, EntityManagerInterface $manager, UserRepository $userRepository, $userId)
+    public function modifyUser(User $user, $userId, UserRepository $repository, EntityManagerInterface $manager)
     {
-        $data = json_decode($request->getContent(), true);
+        $registredUser = $repository->findUser($userId);
+
+        if(empty($registredUser)){
+            return new View('Cet utilisateur n\'existe pas', Response::HTTP_NOT_FOUND);
+        }
+
+        $registredUser=$registredUser[0];
+        $registredUser
+            ->setUsername($user->getUsername())
+            ->setEmail($user->getEmail())
+            ->setPassword($user->getPassword());
 
-        $user = ($userRepository->findUser($userId))[0];
+        $manager->persist($registredUser);
+        $manager->flush();
 
-        if ($user){
-            if($data['username']){
-                $user->setUsername($data['username']);
-            }
-            if ($data['password']){
-                $user->setPassword($data['password']);
-            }
-            if ($data['email']){
-                $user->setEmail($data['email']);
-            }
+        return $this->view($registredUser, Response::HTTP_CREATED);
+    }
 
-            $manager->persist($user);
-            $manager->flush();
+    /**
+     * @Rest\Delete(
+     *     path = "/deleteuser/{userId}",
```

Comment:
la route doit être /users/{id}

Reply:
I can't give the same routes to Delete and Put, can I ?

Reply:
Si si, c'est même le but de rest. Une route pour une même ressource mais avec des VERB (GET, POST, PUT, DELETE) différents

Reply:
Ah ok ! I had not correctly understood this point.
Done